controllers/dspipeline_params.go

```diff
@@ -630,6 +632,57 @@ func (p *DSPAParams) ExtractParams(ctx context.Context, dsp *dspa.DataSciencePip
 			}
 		}
 
+		if cfg := p.APIServer.CustomKfpLauncherConfig; cfg != nil && cfg.Name != "" {
```

Comment:
I would rather break this up and do some proper validation: 
If config is provided but: 
1. it's empty
2. has only one of Name or Datafield
report an error